### PR TITLE
1 login of facilitator cashier and admin

### LINKED ISSUE #1: Login of Facilitator, Cashier, and Admin
**Trigger:**

When the login button is clicked

**Page:**

Image

**Requirements:**

- [ ] Endpoint should be '/login'
- [ ] Used Supabase Auth to login as a user
- [ ] Created 1 user for Facilitator
- [ ] Created 1 used for Cashier
- [ ] Created 1 user for Admin
- [ ] Will return an object of this format:
{
   email: "test@gmail.com",
   userType: "admin"
}

### PR DESCRIPTION
**Pull request by ErriekaP**

### CODE REVIEW

File: src/users/user.controller.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { Controller, Post, Body } from '@nestjs/common';
+import { UserService } from './user.service';
+import { SupabaseService } from 'supabase/supabase.service';
+import { UserDto } from './user.dto';
+import * as bcrypt from 'bcrypt';
+
+@Controller()
+export class UsersController {
+  constructor(
+    private readonly prismaService: UserService,
```
Comment (rhaeyx) [inline src/users/user.controller.ts]: pwede pa rename to userService yung prismaService

File: src/users/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    const prismaData: Prisma.UsersCreateInput = {
+      email: data.email,
+      userType: data.userType,
+      supabaseUserId: user.id
+    };
+    
+
+    return this.prismaService.users.create({
+      data: prismaData,
+    });
```
Comment (rhaeyx) [inline src/users/user.service.ts]: hello, can i ask for the reason for why ganto structure and 
```
    const prismaData: Prisma.UsersCreateInput = {
      email: data.email,
      userType: data.userType,
      supabaseUserId: user.id
    };


    return this.prismaService.users.create({
      data: prismaData,
    });
```
not like this? if no particular reason, pwede pa change ako into this thank u
```
    return await this.prismaService.users.create({
      data: {
        email: data.email,
        userType: data.userType,
        supabaseUserId: user.id,
      },
    });
```

File: src/users/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    private readonly supabaseService: SupabaseService,
+  ) {}
+
+  @Post('register')
```
Comment (rhaeyx) [inline src/users/user.controller.ts]: for testing purposes lang eto na endpoint? if this is used for creating users for the sake of testing, i think a seeder would be more appropriate

File: src/users/user.controller.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { Controller, Post, Body } from '@nestjs/common';
+import { UserService } from './user.service';
+import { SupabaseService } from 'supabase/supabase.service';
+import { UserDto } from './user.dto';
+import * as bcrypt from 'bcrypt';
```
Comment (rhaeyx) [inline src/users/user.controller.ts]: unused import

File: prisma/schema.prisma [hunk 5]
```diff
@@ -65,3 +65,13 @@ model testModel {
   firstName String
   lastName  String
 }
+
+model Users {
```
Comment (rhaeyx) [inline prisma/schema.prisma]: rename Users model to User

https://www.prisma.io/docs/orm/reference/prisma-schema-reference#naming-conventions

File: prisma/migrations/20231228133822_added_users_model/migration.sql [hunk 6]
```diff
@@ -0,0 +1,20 @@
+-- AlterTable
+ALTER TABLE "Event" ADD COLUMN     "requires_payment" BOOLEAN NOT NULL DEFAULT true;
```
Comment (rhaeyx) [inline prisma/migrations/20231228133822_added_users_model/migration.sql]: pwede pa remove ng lines 1 and 2, cause mag error ang migration cause part eto ng iba na migration

File: src/app.module.ts [hunk 7]
```diff
@@ -3,11 +3,14 @@ import { AppController } from './app.controller';
 import { AppService } from './app.service';
 import { StudentModule } from './student/student.module';
 import { PrismaModule } from './prisma/prisma.module';
+import { SupabaseModule } from 'supabase/supabase.module';
+import { AuthModule } from './auth/auth.module';
+import { UsersModule } from './users/user.module';
```
Comment (rhaeyx) [inline src/app.module.ts]: rename to `UsersModule` to `UserModule` please

File: src/prisma/prisma.service.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 import { Injectable } from '@nestjs/common';
-import { PrismaClient } from '@prisma/client';
+import {PrismaClient } from '@prisma/client';
```
Comment (rhaeyx) [inline src/prisma/prisma.service.ts]: add space in between { and PrismaClient

import { PrismaClient } from '@prisma/client';

File: prisma/schema.prisma [hunk 9]
```diff
@@ -65,3 +65,13 @@ model testModel {
...
+  id Int @id @default(autoincrement())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+  supabaseUserId String? @unique
+  email String @unique
+  userType String
+
```
Comment (rhaeyx) [inline prisma/schema.prisma]: please remove extra line

File: supabase/supabase.service.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+import { Injectable } from '@nestjs/common';
+import { createClient, SupabaseClient } from '@supabase/supabase-js';
+
+@Injectable()
+export class SupabaseService {
+  private supabase: SupabaseClient;
+
+  constructor() {
+    const supabaseURL = process.env.SUPABASE_URL;
+    const supabaseAPIKey = process.env.SUPABASE_URL;
```
Comment (rhaeyx) [inline supabase/supabase.service.ts]: shouldnt supabaseAPIKey be the supabase anon key? not sure bakit di nag eerror but either way can you set this to the anon key

**Review by rhaeyx — CHANGES_REQUESTED** [review]
hello, do you have prettier installed? if you dont, i recommend installing prettier also can you add the .env variables to the .env.example template

**Comment by rhaeyx** [comment]
@ErriekaP nag commit ako ng users seeder, you can run it with `npm run prisma:migrate:reset`. pwede mo to gamitin instead of the register na route.

note: this will reset the db, madelete yung old data

File: src/users/user.controller.ts (hunk 3 above)
Comment (ErriekaP) [reply src/users/user.controller.ts]: Hi Francis! Nasali sa seeder ang sa supabase na auth?

Comment (rhaeyx) [reply src/users/user.controller.ts]: yess, `npm run prisma:migrate:reset` lang to seed though matanggal yung previously registered accounts if mag seed

**Comment by viverbungag** [comment]
Already added route guards for all the routes except the creation of students

**Comment by viverbungag** [comment]
Laters nlng nako iactivate ang guards kai basig manlisod ang uban

**Comment by rhaeyx** [comment]
> Laters nlng nako iactivate ang guards kai basig manlisod ang uban

What if mag add lang ta ug 

```
if (process.env.ENVIRONMENT === "dev") { return true } 
``` 
sa canActivate sa guard ya?

edit: process.env.NODE_ENV diay dapat kay naa na daan sa .env.example

**Comment by viverbungag** [comment]
okeokeoke pedee

**Comment by rhaeyx** [comment]
ikaw na merge ani ya @viverbungag ?

**Comment by viverbungag** [comment]
okeokeoke ako na merge ani

**Comment by viverbungag** [comment]
reopen nlng nato kung naai problem

**Pull request merged**